syntax = "proto3";

package marketdata;

message StockRequest {
  string symbol = 1;
}

message StockPrice {
  string symbol = 1;
  double adjustedClose = 2;
  double close = 3;
  double high = 4;
  double low = 5;
  double open = 6;
  int64 volume = 7;
  int64 timestamp_ns = 8; // nanosecond resolution
}

service MarketData {
  rpc Subscribe(StockRequest) returns (stream StockPrice);
}